File: packages/octocode-lsp/benchmark/python/service.py
from typing import Protocol


class Greeter(Protocol):
    def greet(self, name: str) -> str:
        ...


class FriendlyGreeter:
    def greet(self, name: str) -> str:
        return f"Hello, {name}"


def welcome(greeter: Greeter) -> str:
    return greeter.greet("Octocode")

File: packages/octocode-lsp/benchmark/go/service/service.go
package service

type Greeter interface {
	Greet(name string) string
}

type FriendlyGreeter struct{}

func (FriendlyGreeter) Greet(name string) string {
	return "Hello, " + name
}

func Welcome(greeter Greeter) string {
	return greeter.Greet("Octocode")
}
